src/main/java/com/csharks/reportingservice/service/ReportingServiceClass.java

```diff
@@ -43,142 +44,167 @@ public ReportingServiceClass(LeadServiceProxy leadServiceProxy, AccountServicePr
     public List<ReportDTO> reportBySalesRep(String dataType) {
         List<SalesRep> salesReps = getAllSalesReps();
         List<ReportDTO> salesRepReport = new ArrayList<>();
-        if (dataType.toUpperCase().equals("LEAD")) {
+
+        if (dataType.equalsIgnoreCase("LEAD")) {
             List<LeadCountBySalesRepDTO> leads = getCountLeadsBySalesRepId();
             for (LeadCountBySalesRepDTO leadCount : leads) {
-                salesRepReport.add(new ReportDTO(leadCount.getSalesRepId().toString(),leadCount.getLeadCount()));
+                Optional<SalesRep> salesRepByLeadCountId = salesReps.stream()
+                        .filter(salesRep -> salesRep.getId().equals(leadCount.getSalesRepId()))
+                        .findFirst();
+                String salesRepNameByLeadCountId = salesRepByLeadCountId.isPresent() ?
+                        salesRepByLeadCountId.get().getRepName() :
+                        leadCount.getSalesRepId().toString();
+                salesRepReport.add(new ReportDTO(salesRepNameByLeadCountId, leadCount.getLeadCount()));
             }
-        }
-        else if (dataType.toUpperCase().equals("OPPORTUNITY")) {
+        } else if (dataType.equalsIgnoreCase("OPPORTUNITY")) {
             for (SalesRep sales : salesReps) {
                 int count = getBySalesRepId(sales.getId()).size();
-                ReportDTO newReport = new ReportDTO(sales.getRepName(),Long.valueOf(count));
+                ReportDTO newReport = new ReportDTO(sales.getRepName(), Long.valueOf(count));
                 salesRepReport.add(newReport);
             }
-        }
-        else if ((dataType.toUpperCase().equals("CLOSED-WON")) || (dataType.toUpperCase().equals("CLOSED-LOST"))) {
+        } else if ((dataType.replace("-", "_").replace(" ", "-").equalsIgnoreCase("CLOSED_WON")) ||
+                (dataType.replace("-", "_").replace(" ", "-").equalsIgnoreCase("CLOSED_LOST"))) {
             for (SalesRep sales : salesReps) {
-                int count = getBySalesRepIdAndStatus(sales.getId(),dataType).size();
-                ReportDTO newReport = new ReportDTO(sales.getRepName(),Long.valueOf(count));
-                salesRepReport.add(newReport);
+                String formattedDataType = dataType.replace(" ", "_").replace("-", "_").toUpperCase();
+                int count = getBySalesRepIdAndStatus(sales.getId(), formattedDataType).size();
+                salesRepReport.add(new ReportDTO(sales.getRepName(), Long.valueOf(count)));
```

Comment:
For all status calls (OPEN, CLOSED_WON, CLOSED_LOST) added formating and converting. So "open" and "closed-won" should always work on the url